src/plugins/vis_augmenter/public/utils/utils.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { get } from 'lodash';
+import { Vis } from '../../../visualizations/public';
+import {
+  formatExpression,
+  buildExpressionFunction,
+  buildExpression,
+  ExpressionAstFunctionBuilder,
+} from '../../../../plugins/expressions/public';
+import { ISavedAugmentVis, SavedAugmentVisLoader, VisLayerFunctionDefinition } from '../';
+
+// TODO: provide a deeper eligibility check.
+// Tracked in https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3268
+export const isEligibleForVisLayers = (vis: Vis): boolean => {
+  return vis.params.type === 'line';
+};
+
+/**
+ * Using a SavedAugmentVisLoader, fetch all saved objects that are of 'augment-vis' type
+ * and filter out to return the ones associated to the particular vis via
+ * matching vis ID.
+ */
+export const getAugmentVisSavedObjs = async (
+  visId: string | undefined,
+  loader: SavedAugmentVisLoader | undefined
+): Promise<ISavedAugmentVis[]> => {
+  try {
+    const resp = await loader?.findAll();
+    const allSavedObjects = (get(resp, 'hits', []) as any[]) as ISavedAugmentVis[];
+    return allSavedObjects.filter((hit: ISavedAugmentVis) => hit.visId === visId);
```

Comment:
instead of using `findAll`, is there a more specific `find` method we can use (or define on SavedAugmentVisLoader) so that we can just fetch the ones we want?

Reply:
[findall](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/saved_objects/public/saved_object/saved_object_loader.ts#L131) and [find](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/saved_objects/public/saved_object/saved_object_loader.ts#L150) are almost identical, and both allow to pass some query values to narrow down the search. By default the `title` and `description` fields aren't helpful in our use case since we need to search by the vis ID field instead.

But, I do think we could override `findAll()` to include the vis ID field as an option that we can use in the query. Instead of blocking this PR and for the sake of time, can I open a follow-up issue to look into and potentially optimize this call?

Reply:
Created and linked in meta issue enhancement list: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3566